cmd/engine/engine.go

```diff
@@ -244,17 +249,19 @@ func doCmd(_ *cobra.Command, _ []string) (err error) {
 		providers = viper.GetStringSlice(DiscoveryProviderFlag)
 	}
 
-	discoveryService = service_discovery.NewService(
-		service_discovery.WithProviders(providers),
-		service_discovery.WithStorage(db),
-		service_discovery.WithOAuth2Authorizer(
-			// Configure the OAuth 2.0 client credentials for this service
-			&clientcredentials.Config{
-				ClientID:     viper.GetString(ServiceOAuth2ClientIDFlag),
-				ClientSecret: viper.GetString(ServiceOAuth2ClientSecretFlag),
-				TokenURL:     viper.GetString(ServiceOAuth2EndpointFlag),
-			}),
-	)
+	if discoveryService == nil {
```

Comment:
What is the point of this nil check?

Reply:
That the mockDiscoveryService doesn't get over written.